tests/python/pants_test/repo_scripts/test_git_hooks.py

```diff
@@ -0,0 +1,161 @@
+# coding=utf-8
+# Copyright 2014 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import absolute_import, division, print_function, unicode_literals
+
+import datetime
+import os
+import unittest
+from builtins import str
+from contextlib import contextmanager
+
+from pants.util.contextutil import temporary_dir
+from pants.util.dirutil import safe_file_dump, touch
+from pants.util.process_handler import subprocess
+from pants_test.testutils.git_util import get_repo_root, initialize_repo
+
+
+class PreCommitHookTest(unittest.TestCase):
+
+  def setUp(self):
+    self.pants_repo_root = get_repo_root()
+
+  @contextmanager
+  def _create_tiny_git_repo(self):
+    with temporary_dir() as gitdir,\
+         temporary_dir() as worktree:
+      # A tiny little fake git repo we will set up. initialize_repo() requires at least one file.
+      readme_file = os.path.join(worktree, 'README')
+      touch(readme_file)
+      # The contextmanager interface is only necessary if an explicit gitdir is not provided.
+      with initialize_repo(worktree, gitdir=gitdir) as git:
+        yield git, worktree, gitdir
+
+  def _assert_subprocess_error(self, worktree, cmd, expected_excerpt):
+    with self.assertRaises(subprocess.CalledProcessError) as cm:
+      subprocess.check_output(cmd, cwd=worktree)
+    self.assertIn(expected_excerpt, str(cm.exception.output))
+
+  def _assert_subprocess_success(self, worktree, cmd, **kwargs):
+    self.assertEqual(0, subprocess.check_call(cmd, cwd=worktree, **kwargs))
+
+  def _assert_subprocess_success_with_output(self, worktree, cmd, full_expected_output):
+    output = subprocess.check_output(cmd, cwd=worktree)
+    self.assertEqual(full_expected_output, output)
+
+  def _assert_subprocess_error_with_input(self, worktree, cmd, input, expected_excerpt):
+    proc = subprocess.Popen(
+      cmd,
+      cwd=worktree,
+      stdin=subprocess.PIPE,
+      stdout=subprocess.PIPE,
+      stderr=subprocess.PIPE,
+    )
+    (stdout_data, stderr_data) = proc.communicate(input=input)
+    self.assertNotEqual(0, proc.returncode)
+    all_output = '{}\n{}'.format(stdout_data, stderr_data)
+    self.assertIn(expected_excerpt, all_output)
+
+  def test_check_packages(self):
+    package_check_script = os.path.join(self.pants_repo_root, 'build-support/bin/check_packages.sh')
+    with self._create_tiny_git_repo() as (_, worktree, _):
+      init_py_path = os.path.join(worktree, 'subdir/__init__.py')
+
+      # Check that an invalid __init__.py errors.
+      safe_file_dump(init_py_path, 'asdf')
```

Comment:
This function defaults to binary. Add kwarg `mode`
```suggestion
      safe_file_dump(init_py_path, 'asdf', mode='w')
```

Reply:
There are a couple usages of `safe_file_dump` in this file. I'm not going to flag them all, so be sure to command F + command R please.

Reply:
I don't know what that key combination refers to because I use [my own extension for doing horrible text search and replacement things](https://github.com/cosmicexplorer/helm-rg) :)

Reply:
I have added `mode='w'` to all callsites! I'm going to move `pants.base.deprecated` to `pants.util` now so I don't accidentally use `binary_mode` again...